tests/swift-sdk-swift-tests/in-app-tests/InAppPersistenceTests.swift

```diff
@@ -57,4 +57,70 @@ class InAppPersistenceTests: XCTestCase {
             XCTFail("conversion to string failed")
         }
     }
+    
+    func testPersistentReadStateAfterLogout() {
+        let id = "1"
+        
+        let mockInAppFetcher = MockInAppFetcher(messages: [getInboxMessage(id, false)])
+        
+        let internalAPI = IterableAPIInternal.initializeForTesting(inAppFetcher: mockInAppFetcher)
+        let inboxModel = InboxViewControllerViewModel(internalAPIProvider: internalAPI)
+        
+        internalAPI.email = InAppPersistenceTests.email
+        
+        guard let firstMsg = mockInAppFetcher.messages.first else {
+            XCTFail("could not get the first message from fetcher")
+            return
+        }
+        
+        inboxModel.set(read: true, forMessage: inboxModel.message(atIndexPath: IndexPath(row: 0, section: 0)))
+        
+        internalAPI.email = nil
+        
+        internalAPI.email = InAppPersistenceTests.email
+        
+        XCTAssertTrue(firstMsg.read)
+    }
+    
+    func testPersistentReadStateFromServerPayload() {
```

Comment:
This test should initialize with read = false initially, Then have messages arrive from server with read = true and validate that indeed read is true.

Reply:
please see comments below